### PR TITLE
Adding a trait to check for files and folders

### PR DESCRIPTION
**Pull request by burzum**
This is a trait that should be used by every task that uses files and folders as input to check if they really exists.

### CODE REVIEW

File: src/Task/FileSystem/CleanDir.php [hunk 1]
```diff
@@ -16,8 +17,13 @@
  */
 class CleanDir extends BaseDir
 {
+    use FileAndFolderCheck;
+
     public function run()
     {
+        if (!$this->checkResources($this->dirs, 'dir')) {
+            return Result::error($this, 'Resources are missing!');
```
Comment (DavertMik) [inline src/Task/FileSystem/CleanDir.php]: It is hard to get in current context what `resources` are.

File: src/Task/FileSystem/DeleteDir.php [hunk 2]
```diff
@@ -16,8 +17,13 @@
  */
 class DeleteDir extends BaseDir
 {
+    use FileAndFolderCheck;
+
     public function run()
     {
+        if (!$this->checkResources($this->dirs, 'dir')) {
+            return Result::error($this, 'Resources are missing!');
```
Comment (DavertMik) [inline src/Task/FileSystem/DeleteDir.php]: It is hard to get in current context what `resources` are.

Comment (burzum) [inline src/Task/FileSystem/DeleteDir.php]: Suggest a better name?

File: src/Task/FileSystem/CleanDir.php (hunk 1 above)
Comment (burzum) [inline src/Task/FileSystem/CleanDir.php]: Suggest a better name?

File: src/Common/FileAndFolderCheck.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
+<?php
+namespace Robo\Common;
+
+trait FileAndFolderCheck
```
Comment (DavertMik) [inline src/Common/FileAndFolderCheck.php]: strange that it is named `FileAndFolder` while the parameter used is `fileAndDir`. I'd rename it to `FileExistenceChecker` or `ResourceExistenceChecker`

**Comment by DavertMik** [comment]
Thanks, I've got some comments

File: src/Task/FileSystem/CleanDir.php (hunk 1 above)
Comment (DavertMik) [inline src/Task/FileSystem/CleanDir.php]: For this specific case the error should state like "Source directories are missing"

**Comment by DavertMik** [comment]
Wow, great work! Especially with tests and improving current tasks. 
Is that ready, or some other tasks should be refactored to use it?

**Comment by burzum** [comment]
It's ready. I think there are no other tasks left.

**Comment by boedah** [comment]
Just linking #231

**Comment by DavertMik** [comment]
The day has come!

**Pull request merged**